src/shared/components/deletion-confirmation/DeletionConfirmation.test.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import DeletionConfirmation, {
+  DeletionConfirmationProps
+} from './DeletionConfirmation';
+
+const props: DeletionConfirmationProps = {
+  onConfirm: jest.fn(),
+  onCancel: jest.fn()
+};
+
+describe('DeletionConfirmation', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders with default message', () => {
+    const { container } = render(<DeletionConfirmation {...props} />);
+
+    const deleteButton = container.querySelector('button') as HTMLButtonElement;
+    expect(deleteButton?.textContent).toBe('Delete');
```

Comment:
What's the warning message? What's the cancel message? Why are we checking specifically the button message, but not the other two?

Reply:
The warning message or the cancel message can be anything, why are we checking for the text? Should we not be checking if the component renders. 
I am a bit confused as to why we need to test these default text. i thought we dont check the exact text in a component?

Reply:
>  why are we checking for the text

I don't know. Why are you checking the text of the delete button? It can be anything as well.